app/client/src/components/app.tsx

```diff
@@ -14,35 +15,124 @@ import "uswds/js/uswds.js";
 // ---
 import { serverBasePath, serverUrl, fetchData } from "../config";
 import Loading from "components/loading";
+import ConfirmationDialog from "components/confirmationDialog";
 import Welcome from "components/welcome";
 import Dashboard from "components/dashboard";
 import AllRebateForms from "routes/allRebateForms";
 import NewRebateForm from "routes/newRebateForm";
 import ExistingRebateForm from "routes/existingRebateForm";
 import NotFound from "routes/notFound";
 import { useUserState, useUserDispatch } from "contexts/user";
+import { Action, useDialogDispatch, useDialogState } from "contexts/dialog";
 
 function ProtectedRoute({ children }: { children: JSX.Element }) {
   const { pathname } = useLocation();
-  const { isAuthenticating, isAuthenticated } = useUserState();
-  const dispatch = useUserDispatch();
+  const { epaUserData, isAuthenticating, isAuthenticated } = useUserState();
+  const userDispatch = useUserDispatch();
 
-  // check if user is already logged in or needs to be redirected to /welcome route
-  useEffect(() => {
-    dispatch({ type: "FETCH_EPA_USER_DATA_REQUEST" });
+  // Check if user is already logged in or needs to be redirected to /welcome route
+  const checkUser = useCallback(() => {
     fetchData(`${serverUrl}/api/v1/epa-data`)
       .then((res) => {
-        dispatch({
+        userDispatch({
           type: "FETCH_EPA_USER_DATA_SUCCESS",
           payload: { epaUserData: res },
         });
-        dispatch({ type: "USER_SIGN_IN" });
+        userDispatch({ type: "USER_SIGN_IN" });
       })
       .catch((err) => {
-        dispatch({ type: "FETCH_EPA_USER_DATA_FAILURE" });
-        dispatch({ type: "USER_SIGN_OUT" });
+        userDispatch({ type: "FETCH_EPA_USER_DATA_FAILURE" });
+        userDispatch({ type: "USER_SIGN_OUT" });
       });
-  }, [dispatch, pathname]);
+  }, [userDispatch]);
+
+  useEffect(() => {
+    userDispatch({ type: "FETCH_EPA_USER_DATA_REQUEST" });
```

Comment:
@courtneymyers I found in testing that if I don't call the `FETCH_EPA_USER_DATA_REQUEST` action during the inactivity refreshes, the children components do not re-render, so the user will not lose any form progress during the refresh. Because of this workaround, I kept this logic under ProtectedRoutes, but I can still move if you think there's a better place for it.

Reply:
Smart! This is a great solution. When I first read this from the email notification I hadn't looked at the code yet...I assumed you weren't calling `FETCH_EPA_USER_DATA_REQUEST` anymore, which would have been a concern. But I see you moved it in to the `useEffect` (and the checking of user auth into a memoized callback) so the `FETCH_EPA_USER_DATA_REQUEST` action is still dispatched on every route change, so that's great.